File: vipclaw-admin/src/main/java/com/vipclaw/admin/entity/SysUser.java
package com.vipclaw.admin.entity;

import com.baomidou.mybatisplus.annotation.*;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.io.Serial;
import java.io.Serializable;
import java.time.LocalDateTime;

/**
 * 用户实体类
 *
 * @author vipamp
 * @since 2026-03-05
 */
@Data
@EqualsAndHashCode(callSuper = false)
@TableName("sys_user")
@Schema(description = "用户实体类")
public class SysUser implements Serializable {

    @Serial
    private static final long serialVersionUID = 1L;

    /**
     * 用户 ID
     */
    @TableId(value = "id", type = IdType.AUTO)
    @Schema(description = "用户 ID")
    private Long id;

    /**
     * 用户名
     */
    @Schema(description = "用户名")
    private String username;

    /**
     * 密码
     */
    @Schema(description = "密码")
    private String password;

    /**
     * 昵称
     */
    @Schema(description = "昵称")
    private String nickname;

    /**
     * 邮箱
     */
    @Schema(description = "邮箱")
    private String email;

    /**
     * 手机号
     */
    @Schema(description = "手机号")
    private String phone;

    /**
     * 性别 (0:女 1:男 2:未知)
     */
    @Schema(description = "性别 (0:女 1:男 2:未知)")
    private Integer gender;

    /**
     * 头像 URL
     */
    @Schema(description = "头像 URL")
    private String avatar;

    /**
     * 是否启用（0:禁用，1:启用）
     */
    @Schema(description = "是否启用（0:禁用，1:启用）")
    private Integer status;

    /**
     * 是否可用（0:被删除，1:可用）
     */
    @Schema(description = "是否可用（0:被删除，1:可用）")
    @TableLogic(value = "1", delval = "0")
    private Integer active;

    /**
     * 创建时间
     */
    @Schema(description = "创建时间")
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    /**
     * 更新时间
     */
    @Schema(description = "更新时间")
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;
}

File: vipclaw-admin/src/main/kotlin/com/vipamp/vipclaw/admin/entity/SysUser.kt
package com.vipamp.vipclaw.admin.entity

import io.swagger.v3.oas.annotations.media.Schema
import java.io.Serializable
import java.time.LocalDateTime

@Schema(description = "User entity")
class SysUser : Serializable {
    companion object {
        private const val serialVersionUID = 1L
    }

    @Schema(description = "User ID")
    var id: Long = 0

    @Schema(description = "Tenant ID (primary tenant)")
    var tenantId: Long? = null

    @Schema(description = "Username")
    var username: String = ""

    @Schema(description = "Password")
    var password: String = ""

    @Schema(description = "Nickname")
    var nickname: String = ""

    @Schema(description = "Email")
    var email: String = ""

    @Schema(description = "Phone number")
    var phone: String = ""

    @Schema(description = "Gender (0:female, 1:male, 2:unknown)")
    var gender: Int = 2

    @Schema(description = "Avatar URL")
    var avatar: String? = null

    @Schema(description = "Status (0:disabled, 1:enabled)")
    var status: Int = 1

    @Schema(description = "Admin status (0:no, 1:yes)")
    var isAdmin: Int = 0

    @Schema(description = "Last login time")
    var lastLoginTime: LocalDateTime? = null

    @Schema(description = "Active status (0:deleted, 1:active)")
    var active: Int = 1

    @Schema(description = "Creation time")
    var createTime: LocalDateTime = LocalDateTime.now()

    @Schema(description = "Update time")
    var updateTime: LocalDateTime = LocalDateTime.now()
}
